diesel_codegen/src/embed_migrations.rs

```diff
@@ -16,7 +16,7 @@ pub fn derive_embed_migrations(input: syn::MacroInput) -> quote::Tokens {
 
     let options = get_options_from_input(&input.attrs, bug);
     let migrations_path_opt = options.map(|o| get_option(&o, "migrations_path", bug));
```

Comment:
I think the problem is that `.map` creates a reference that is only valid in its closure, but you want to return a value that lives as long as `option` does (and not the reference to option). Try this:

```diff
- let migrations_path_opt = options.map(|o| get_option(&o, "migrations_path", bug));
+ let migrations_path_opt = options.as_ref().map(|o| get_option(o, "migrations_path", bug));
```